### DISCUSSION TITLE
How to use several clients?

### DISCUSSION THREAD

**Discussion by mogikanin** [getsentry/sentry-dotnet#3183] [General]
Hi, could you say what's the correct way to use several independent sentry clients with different settings (dsn) inside one app? Didn't find anything similar inside your samples... you are using static `SentrySdk.Capture()` everywhere.

Thanks in advance!

**Comment by bruno-garcia** [comment]
You can bind a new client to the current scope: https://github.com/getsentry/sentry-dotnet/blob/c19fb14d5c2a56cbe162f1ef5e30af4a8149b5fa/src/Sentry/SentrySdk.cs#L294

Mind if I ask what's the use case? You can also keep a reference of the Client and call directly to it to capture errors and stuff but some functionality go to the client bound to the scope

**Comment by mogikanin** [comment]
@bruno-garcia thanks, it works fine. We use several sentry projects for different purposes. One project (per app) - for internal bugs, warnings, etc... Another project - for collecting users feedback, and this project is shared across our apps to have all feedbacks in one place. So, from single app we need possibility to capture events into different sentry projects. 

One note, it seems there is no way to **Unbind** client when we don't need it anymore?

Here the sample if someone will be needed in similar functionality:
```
// Setup and bind client1
var client1 = new SentryClient(new SentryOptions{/*put options*/});
SentrySdk.BindClient(client1);

// Setup and bind client2
var client2 = new SentryClient(new SentryOptions{/*put options*/});
SentrySdk.BindClient(client2);

//.. Later in code
client1.CaptureEvent(....)
client2.CaptureEvent(....)
```

**Comment by bitsandfoxes** [comment]
If I get your intention correctly you're looking for something like this
```
using (SentrySdk.PushScope())
{
    var client1 = new SentryClient(new SentryOptions());
    SentrySdk.BindClient(client1);
}
```

The client gets bound to the current scope. Pushing a scope gives you an `IDisposable` that will pop the scope-client pair from the stack when the scope gets disposed.

**Comment by mogikanin** [comment]
@bitsandfoxes not obvious solution since we don't use explicitly result from `SentrySdk.PushScope()`, but ok. Also, it is worth to mention that `IsGlobalModeEnabled` should be disabled in options, [right](https://github.com/getsentry/sentry-dotnet/blob/main/src/Sentry/Internal/SentryScopeManager.cs#L51)? 
Also, it will work correctly in next scenario, or I have to take care about order of pushed scopes-clients?
```
public void Init()
{
    _scope1 = SentrySdk.PushScope();
    _client1 = new SentryClient(new SentryOptions{});
    SentrySdk.BindClient(_client1);
    
    _scope2 = SentrySdk.PushScope();
    _client2 = new SentryClient(new SentryOptions{});
    SentrySdk.BindClient(_client2);
}


//-------- Later, in other method

// we don't need client1 anymore, but still need client2
_client1.Dispose();
_scope1.Dispose();
```

**Comment by bitsandfoxes** [comment]
Yes, `GlobalModeEnabled` needs to be disabled.
I don't think the snippet would work in the way you would expect it to. We've got a `<Scope, Client>` stack going on, and `BindClient` binds the provided client to the currently active scope so when you call into the static API your bound client does the work. The client gets unbound by the scope getting popped off the stack on dispose. Pushing onto the scope twice, the current one is the last one added.
